Makefile

```diff
@@ -1,21 +1,44 @@
-GO=CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go
-TAG=0.1.4
-BIN=aws-limits-exporter
-IMAGE=danielfm/$(BIN)
+.PHONY: test
+export GOBIN := $(PWD)/bin
+export PATH := $(GOBIN):$(PATH)
+export INSTALL_FLAG=
+export TAG=0.2.0
+
+DOCKER_IMAGE = aws-limits-exporter
+DOCKER_REPO = danielfm/
+
+# Determine which OS.
+OS?=$(shell uname -s | tr A-Z a-z)
+
+default: build
+
+dependencies:
+	@go mod tidy -v
+
+dep: dependencies
+
+run-server: build
+	$(GOBIN)/aws-limits-exporter
+
+run-linux: build
+	$(GOBIN)/aws-limits-exporter
+
+test:
+	@go test ./... -timeout 2m -v -race
+
+test-cover:
+	@go test ./... -timeout 2m -race -cover
 
-.PHONY: build
 build:
-	$(GO) build -a --ldflags "-X main.VERSION=$(TAG) -w -extldflags '-static'" -tags netgo -o bin/$(BIN) ./cmd
+	CGO_ENABLED=0 GOOS=$(OS) go build $(INSTALL_FLAG) -a --ldflags "-X main.VERSION=$(TAG) -w -extldflags '-static'" -tags netgo -o $(GOBIN)/aws-limits-exporter ./cmd
 
-.PHONY: image
-image: build
-	docker build -t $(IMAGE):$(TAG) .
+clean:
+	@go clean
 
-.PHONY: push
-push: image
-	docker push $(IMAGE):$(TAG)
+docker-build:
+	docker build -t ${DOCKER_REPO}$(DOCKER_IMAGE):latest .
```

Comment:
Does this `${DOCKER_REPO}` interpolate correctly, or should we change it to `$(DOCKER_REPO)`? The same applies to the commands in `docker-deploy` target.

Reply:
it worked, however I changed the commands to add the `/`, so the repo name can be let alone. Ex: 
`make docker-deploy
docker tag danielfm/aws-limits-exporter:latest danielfm/aws-limits-exporter:0.2.0
docker push danielfm/aws-limits-exporter:0.2.0`